src/app/model/ChessEngine.ts

```diff
@@ -99,8 +111,107 @@ export class ChessEngine implements ChessModel {
     }
   }
 
+  private intendToCastling(squareFrom: Square, squareTo: Square): boolean {
+    const chosenPiece = this.board.onPositionPiece(squareFrom);
+    const whiteCastlingFrom: Square = { column: 'E', row: 1 };
+    const whiteLongCastlingTo: Square = { column: 'C', row: 1 };
+    const whiteShortCastlingTo: Square = { column: 'G', row: 1 };
+    const blackCastlingFrom: Square = { column: 'E', row: 8 };
+    const blackLongCastlingTo: Square = { column: 'C', row: 8 };
+    const blackShortCastlingTo: Square = { column: 'G', row: 8 };
+
+    return (
+      chosenPiece instanceof King &&
+      ((chosenPiece.side === Side.WHITE &&
+        this.isSameSquare(squareFrom, whiteCastlingFrom) &&
+        (this.isSameSquare(squareTo, whiteLongCastlingTo) || this.isSameSquare(squareTo, whiteShortCastlingTo))) ||
+        (chosenPiece.side === Side.BLACK &&
+          this.isSameSquare(squareFrom, blackCastlingFrom) &&
+          (this.isSameSquare(squareTo, blackLongCastlingTo) || this.isSameSquare(squareTo, blackShortCastlingTo))))
+    );
+  }
+
+  private isSameSquare(squareA: Square, squareB: Square) {
+    return squareA.column === squareB.column && squareA.row === squareB.row;
+  }
+
+  private isCastlingPossible(kingSquareFrom: Square, kingSquareTo: Square): boolean {
+    let squareCrossedByKing!: Square;
+    let rookPosition!: Square;
+    let moveDirection!: number;
+    switch (kingSquareTo.column) {
+      case 'C':
+        squareCrossedByKing = { column: 'D', row: kingSquareTo.row };
+        rookPosition = { column: 'A', row: kingSquareTo.row };
+        moveDirection = -1;
+        break;
+      case 'G':
+        squareCrossedByKing = { column: 'F', row: kingSquareTo.row };
+        rookPosition = { column: 'H', row: kingSquareTo.row };
+        moveDirection = 1;
+        break;
+    }
+    return !(
+      this.isKingChecked(this.board, this.currentSide) ||
+      this.isSquareChecked(this.board, this.currentSide, squareCrossedByKing) ||
+      this.willBeKingChecked(kingSquareFrom, kingSquareTo) ||
+      this.isAnyPieceBetweenKingAndRook(kingSquareFrom, rookPosition, moveDirection) ||
+      this.pieceWasAlreadyMoved(kingSquareFrom) ||
+      this.pieceWasAlreadyMoved(rookPosition)
+    );
+  }
+
+  private isAnyPieceBetweenKingAndRook(actualPosition: Square, rookPosition: Square, moveDirection: number): boolean {
+    const nextSquare: Square = {
+      column: columns[columns.indexOf(actualPosition.column) + moveDirection],
+      row: actualPosition.row,
+    };
+    if (nextSquare.column === rookPosition.column) {
+      return false;
+    }
+    const isSquareOccupied = this.board.onPositionPiece(nextSquare);
+    return isSquareOccupied ? true : this.isAnyPieceBetweenKingAndRook(nextSquare, rookPosition, moveDirection);
+  }
+
+  private pieceWasAlreadyMoved(squareWithPiece: Square) {
```

Comment:
To jest square bez piece :)

Reply:
Yy, co proszę?

Reply:
Bo masz nazwe: squareWithPiece

Reply:
Nawet jak wiemy, że coś tam na nim stoi? :P

Reply:
Dobra, nie zrozumialem tej metody :)